source/geometry_model/box.cc

```diff
@@ -257,6 +259,163 @@ namespace aspect
       return std::min (std::max (d, 0.), maximal_depth());
     }
 
+    template <int dim>
+    void
+    Box<dim>::
+    update ()
+    {
+        AssertThrow(Plugins::plugin_type_matches<const GeometryModel::Box<dim>>(this->get_geometry_model()),
+                  ExcMessage("The geometry update function currently only works with a 2D box geometry model."));
+
+        AssertThrow(dim==2,
+                  ExcMessage("The geometry update function currently only works with a 2D box geometry model."));
+
+        this->get_pcout() << "   Updating surface values... " <<std::endl;
+        TimerOutput::Scope timer_section(this->get_computing_timer(), "Geometry model surface update");
+
+        // loop over all of the surface cells and save the elevation to a stored value.
+        // This needs to be sent to 1 processor, sorted, and broadcast so that every processor knows the entire surface.
+        // (Does bcast create memory of the entire variable on every processor or only a pointer to the one stored on 0?)
+        std::vector<std::vector<double>> local_surface_height;
+        Table<dim-1, double> data_table;
+        const types::boundary_id relevant_boundary = this->get_geometry_model().translate_symbolic_boundary_name_to_id ("top");
+        const QTrapezoid<dim-1> face_corners;
+        FEFaceValues<dim> fe_face_values(this->get_mapping(),
+                                          this->get_fe(),
+                                          face_corners,
+                                          update_quadrature_points);
+
+        // Loop over all corners at the surface and save their position.
+        // TODO: Update this to work in 3D. Spherical?
+        for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+          if (cell->is_locally_owned() && cell->at_boundary())
+            for (const unsigned int face_no : cell->face_indices())
+              if (cell->face(face_no)->at_boundary())
+                {
+                  if ( cell->face(face_no)->boundary_id() != relevant_boundary)
+                    continue;
+
+                  fe_face_values.reinit(cell, face_no);
+
+                  for (unsigned int corner = 0; corner < face_corners.size(); ++corner)
+                    {
+                      const Point<dim> vertex = fe_face_values.quadrature_point(corner);
+
+                      // We can't push back a point so we convert it into a vector.
```

Comment:
I dont understand this comment. The following should work perfectly fine:
```
std::vector<Point<dim>> points;
Point<dim> point;
points.push_back(point);
```

Can you explain the problems you had in a bit more detail? I am sure we can figure out how to make this work with a vector of points.

Reply:
You're right this does work I don't what the comment was for. I did run into an issue later when trying to use compute_set_union with "std::vector<Point<dim>> points;" should I leave it as a vector of vectors for this?

Here is an example of the error message:

> required from 'std::vector<ValueType> dealii::Utilities::MPI::compute_set_union(const >std::vector<ValueType>&, MPI_Comm) [with T = dealii::Point<2, double>; MPI_Comm = ompi_communicator_t*]'
>/home/bbpdneu1/software/depth_mesh/aspect/source/geometry_model/box.cc:327:45:   required from 'void aspect::GeometryModel::Box<dim>::update() [with int dim = 2]'
/home/bbpdneu1/software/depth_mesh/aspect/source/geometry_model/box.cc:609:5:   required from here
/sw/spack/24.05/linux-rocky8-x86_64/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/gcc-13.2.0-ywo3vxag72klu2wt4of6ot7kw7mftyig/lib/gcc/x86_64-pc-linux-gnu/13.2.0/../../../../include/c++/13.2.0/bits/predefined_ops.h:69:22: error: no match for 'operator<' (operand types are 'dealii::Point<2, double>' and 'dealii::Point<2, double>')
   69 |       { return *__it < __val; }